Branch-SDK/src/io/branch/referral/IntegrationValidator.java

```diff
@@ -0,0 +1,286 @@
+package io.branch.referral;
+
+import android.app.Activity;
+import android.app.ActivityManager;
+import android.app.AlertDialog;
+import android.content.ActivityNotFoundException;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Build;
+import android.provider.Browser;
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.jar.JarFile;
+
+/**
+ * Created by sojanpr on 9/15/17.
+ */
+
+public class IntegrationValidator {
+
+    public void validateSDKIntegration (Context context){
```

Comment:
Do you think it is good to do all this logic in background thread. Preferably on  a Async Task with some progress updates

Reply:
Synced offline. Since this call is not meant to be made on the production deployment we should be okay with this.